### PR TITLE
Gm/no fork lint fix

### PR DESCRIPTION
**Pull request by guym-blox**

### CODE REVIEW

**Comment by y0sher** [comment]
There are still lint errors if you run `make lint` or look at CI lint report.
they are not related to spec tests.

@guym-blox 

```
INFO [runner] linters took 14.88702675s with stages: goanalysis_metalinter: 14.744550208s 
network/topics/container.go:1: : # github.com/ssvlabs/ssv/network/topics [github.com/ssvlabs/ssv/network/topics.test]
network/topics/controller_test.go:70:27: undefined: validation.NewMessageValidator
network/topics/controller_test.go:375:45: not enough arguments in call to NewPubSub
        have (context.Context, *zap.Logger, *PubSubConfig, metricsreporter.MetricsReporter)
        want (context.Context, *zap.Logger, *PubSubConfig, Metrics, CommitteesProvider)
network/topics/controller_test.go:425:19: undefined: specqbftSignedMessage
network/topics/msg_validator_test.go:53:19: undefined: validation.NewMessageValidator
network/topics/msg_validator_test.go:53:77: undefined: validation.WithNodeStorage
network/topics/msg_validator_test.go:92:39: undefined: spectypes.EncodeSignedSSVMessage
network/topics/msg_validator_test.go:93:54: ssvMsg.GetID().GetPubKey undefined (type "github.com/ssvlabs/ssv-spec/types".MessageID has no field or method GetPubKey) (typecheck)
package topics
protocol/v2/testing/test_utils.go:121:18: Error return value of `file.Close` is not checked (errcheck)
        defer file.Close()
                        ^
protocol/v2/testing/test_utils.go:127:24: Error return value of `gzipReader.Close` is not checked (errcheck)
        defer gzipReader.Close()
                              ^
protocol/v2/testing/test_utils.go:280:17: Error return value of `outFile.Close` is not checked (errcheck)
                        outFile.Close()
                                     ^
protocol/v2/testing/test_utils.go:295:14: Error return value of `errors.Wrap` is not checked (errcheck)
                errors.Wrap(err, "could not open file")
                           ^
protocol/v2/ssv/validator/committee.go:145:19: Error return value of `c.ConsumeQueue` is not checked (errcheck)
        go c.ConsumeQueue(logger, duty.Slot, c.ProcessMessage, c.Runners[duty.Slot])
                         ^
protocol/v2/testing/test_utils.go:117:15: G304: Potential file inclusion via variable (gosec)
        file, err := os.Open(gzPath)
                     ^
protocol/v2/testing/test_utils.go:269:14: G301: Expect directory permissions to be 0750 or less (gosec)
                        if err := os.Mkdir(header.Name, 0755); err != nil {
                                  ^
protocol/v2/testing/test_utils.go:277:17: G110: Potential DoS vulnerability via decompression bomb (gosec)
                        if _, err := io.Copy(outFile, tarReader); err != nil {
                                     ^
protocol/v2/ssv/spectest/debug_states.go:52:8: G306: Expect WriteFile permissions to be 0600 or less (gosec)
        err = os.WriteFile(fmt.Sprintf("%s/%s_test_serialized.json", dumpDir, name), bytes, 0644)
              ^
protocol/v2/testing/test_utils.go:292:6: func `unpackTestsJson` is unused (unused)
func unpackTestsJson(path string) error {
     ^
protocol/v2/qbft/instance/instance.go:30:2: field `started` is unused (unused)
        started time.Time
        ^
protocol/v2/qbft/instance/instance.go:66:6: func `messageIDFromBytes` is unused (unused)
func messageIDFromBytes(mid []byte) spectypes.MessageID {
     ^
message/validation/consensus_validation.go:333:29: func `(*messageValidator).maxSlotsInState` is unused (unused)
func (mv *messageValidator) maxSlotsInState() phase0.Slot {
                            ^
protocol/v2/ssv/validator/committee.go:320:21: func `(*Committee).updateAttestingSlotMap` is unused (unused)
func (c *Committee) updateAttestingSlotMap(duty *spectypes.CommitteeDuty) {
                    ^
operator/validator/controller.go:194:2: field `recentlyStartedCommittees` is unused (unused)
        recentlyStartedCommittees uint64
        ^
operator/validator/controller.go:893:6: type `starter` is unused (unused)
type starter interface {
     ^
protocol/v2/ssv/spectest/msg_processing_type.go:172:36: func `(*MsgProcessingSpecTest).compareBroadcastedBeaconMsgs` is unused (unused)
func (test *MsgProcessingSpecTest) compareBroadcastedBeaconMsgs(t *testing.T) {
                                   ^
protocol/v2/testing/test_utils.go:256:2: S1006: should use for {} instead of for true {} (gosimple)
        for true {
        ^
network/topics/params/message_rate.go:113:5: S1009: should omit nil check; len() for []*github.com/ssvlabs/ssv/registry/storage.Committee is defined as zero (gosimple)
        if committees == nil || len(committees) == 0 {
           ^
protocol/v2/ssv/testing/runner.go:110:10: S1040: type assertion to the same type: runner.NewCommitteeRunner(
        networkconfig.TestNetwork,
        shareMap,
        contr,
        tests.NewTestingBeaconNodeWrapped(),
        net,
        km,
        opSigner,
        valCheck,
) already has type runner.Runner (gosimple)
                return runner.NewCommitteeRunner(
                       ^
protocol/v2/ssv/testing/runner.go:121:10: S1040: type assertion to the same type: runner.NewAggregatorRunner(
        spectypes.BeaconTestNetwork,
        shareMap,
        contr,
        tests.NewTestingBeaconNodeWrapped(),
        net,
        km,
        opSigner,
        valCheck,
        TestingHighestDecidedSlot,
) already has type runner.Runner (gosimple)
                return runner.NewAggregatorRunner(
                       ^
protocol/v2/ssv/testing/runner.go:133:10: S1040: type assertion to the same type: runner.NewProposerRunner(
        spectypes.BeaconTestNetwork,
        shareMap,
        contr,
        tests.NewTestingBeaconNodeWrapped(),
        net,
        km,
        opSigner,
        valCheck,
        TestingHighestDecidedSlot,
) already has type runner.Runner (gosimple)
                return runner.NewProposerRunner(
                       ^
protocol/v2/ssv/testing/runner.go:145:10: S1040: type assertion to the same type: runner.NewSyncCommitteeAggregatorRunner(
        spectypes.BeaconTestNetwork,
        shareMap,
        contr,
        tests.NewTestingBeaconNodeWrapped(),
        net,
        km,
        opSigner,
        valCheck,
        TestingHighestDecidedSlot,
) already has type runner.Runner (gosimple)
                return runner.NewSyncCommitteeAggregatorRunner(
                       ^
protocol/v2/ssv/testing/runner.go:157:10: S1040: type assertion to the same type: runner.NewValidatorRegistrationRunner(
        spectypes.BeaconTestNetwork,
        shareMap,
        contr,
        tests.NewTestingBeaconNodeWrapped(),
        net,
        km,
        opSigner,
) already has type runner.Runner (gosimple)
                return runner.NewValidatorRegistrationRunner(
                       ^
protocol/v2/ssv/testing/runner.go:167:10: S1040: type assertion to the same type: runner.NewVoluntaryExitRunner(
        spectypes.BeaconTestNetwork,
        shareMap,
        tests.NewTestingBeaconNodeWrapped(),
        net,
        km,
        opSigner,
) already has type runner.Runner (gosimple)
                return runner.NewVoluntaryExitRunner(
                       ^
protocol/v2/ssv/testing/runner.go:187:10: S1040: type assertion to the same type: ret already has type runner.Runner (gosimple)
                return ret.(runner.Runner)
                       ^
operator/validator/controller.go:936:8: lostcancel: the cancel function returned by context.WithCancel should be called, not discarded, to avoid a context leak (govet)
                ctx, _ := context.WithCancel(c.context)
                     ^
protocol/v2/testing/test_utils.go:283:4: printf: log.Fatalf format %s has arg header.Typeflag of wrong type byte (govet)
                        log.Fatalf(
                        ^
protocol/v2/ssv/validator/non_committee_validator.go:86:10: printf: fmt.Errorf format %s has arg partialSigMessages.Type of wrong type github.com/ssvlabs/ssv-spec/types.PartialSigMsgType (govet)
                return fmt.Errorf("not processing message type %s", partialSigMessages.Type)
                       ^
protocol/v2/ssv/spectest/msg_processing_type.go:137:2: ineffectual assignment to beaconNetwork (ineffassign)
        beaconNetwork := tests.NewTestingBeaconNodeWrapped()
        ^
eth/eventhandler/handlers.go:297:15: SA4010: this result of append is never used, except maybe in other appends (staticcheck)
                committee = append(committee, &spectypes.CommitteeMember{
                            ^
eth/eventhandler/handlers.go:332:15: SA4010: this result of append is never used, except maybe in other appends (staticcheck)
                operators = append(operators, &spectypes.Operator{
                            ^
integration/qbft/tests/temp_testing_beacon_network.go:21:2: SA5007: infinite recursive call (staticcheck)
        bn.SetSyncCommitteeAggregatorRootHexes(roots)
        ^
```

**Review by y0sher — CHANGES_REQUESTED** [review]
fix rest of lint errs

File: protocol/v2/qbft/spectest/msg_processing_type.go [hunk 1]
```diff
@@ -31,7 +32,7 @@ func RunMsgProcessing(t *testing.T, test *spectests.MsgProcessingSpecTest) {
 	msgId := specqbft.ControllerIdToMessageID(test.Pre.State.ID)
 	logger := logging.TestLogger(t)
 	pre := instance.NewInstance(
-		qbfttesting.TestingConfig(logger, spectestingutils.KeySetForCommitteeMember(test.Pre.State.CommitteeMember), msgId.GetRoleType()),
+		qbfttesting.TestingConfig(logger, spectestingutils.KeySetForCommitteeMember(test.Pre.State.CommitteeMember), convert.RunnerRole(msgId.GetRoleType())),
```
Comment (y0sher) [inline protocol/v2/qbft/spectest/msg_processing_type.go:35]: `TestingConfig` should receive `spectypes.RunnerRole`, use the `convert.RunnerRole` only where required (db and exporter).
its like this in `alan/no-fork` already

File: protocol/v2/ssv/validator/committee.go [hunk 2]
```diff
@@ -148,7 +148,12 @@ func (c *Committee) StartDuty(logger *zap.Logger, duty *spectypes.CommitteeDuty)
 
 	logger = c.logger.With(fields.DutyID(fields.FormatCommitteeDutyID(c.Operator.Committee, c.BeaconNetwork.EstimatedEpochAtSlot(duty.Slot), duty.Slot)), fields.Slot(duty.Slot))
 	// TODO alan: stop queue
-	go c.ConsumeQueue(logger, duty.Slot, c.ProcessMessage, c.Runners[duty.Slot])
+	go func() {
+		err := c.ConsumeQueue(logger, duty.Slot, c.ProcessMessage, c.Runners[duty.Slot])
+		if err != nil {
+			logger.Warn("handles error message", zap.Error(err))
```
Comment (y0sher) [inline protocol/v2/ssv/validator/committee.go:154]:
```suggestion
			logger.Error("❗ failed consuming committee queue", zap.Error(err))
```

**Review by y0sher — COMMENTED** [review]
@pavelkrolevets please review the p2p tests code

**Comment by y0sher** [comment]
@guym-blox , the new p2p testing code lies here - https://github.com/ssvlabs/ssv/pull/1442 
if it passes we can use it here as well

**Comment by pavelkrolevets** [comment]
> @guym-blox , the new p2p testing code lies here - #1442 if it passes we can use it here as well. Edit, from my last check, they do not pass, @pavelkrolevets can you confirm?

Checked p2p tests locally are passing `[alan/no-fork-fix-tests](https://github.com/ssvlabs/ssv/tree/alan/no-fork-fix-tests)`. Try to clean cache `go clean -cache`. Please let me know which test is failing Ill check it again.

File: network/p2p/p2p_test.go [hunk 3]
```diff
@@ -33,13 +39,12 @@ func TestP2pNetwork_SubscribeBroadcast(t *testing.T) {
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 
-	ks := spectestingutils.Testing4SharesSet()
 	pks := []string{"8e80066551a81b318258709edaf7dd1f63cd686a0e4db8b29bbb7acfe65608677af5a527d9448ee47835485e02b50bc0"}
-	ln, routers, err := CreateNetworkAndSubscribeFromKeySet(t, ctx, LocalNetOptions{
```
Comment (pavelkrolevets) [inline network/p2p/p2p_test.go:38]: You can reuse test from here https://github.com/ssvlabs/ssv/pull/1442

**Comment by y0sher** [comment]
@pavelkrolevets since this PR is only taking care of Lint, please take care of the p2p tests in your branch.

**Pull request merged**